hal/test-headers/rtc_api_tests.h

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
```

Comment:
We don't usually have test headers, maybe it would make more sense to pull all the requirements to a main API header and porting guide and test case descriptions to the test c file? I think having yet another header to look at is a bit confusing.

Reply:
This is required to allow verification tests to be referenced from the doxygen.